File: FoodHub Web API/Services/AccountService.cs
namespace FoodHub_Web_API.Services
{
    public interface IAccountService
    {
        Task<List<StaticAccountResponse>> GetAll();
        Task<DirectAccountResponse> GetById(int accountId);
        Task<DirectAccountResponse> Update(int accountId, AccountRequest request);
    }

    public class AccountService
    {

        private readonly IAccountRepository _accountRepository;
        private readonly IMapper _mapper;

        public AccountService(IAccountRepository accountRepository, IMapper mapper)
        {
            _accountRepository = accountRepository;
            _mapper = mapper;
        }

        public async Task<List<StaticAccountResponse>> GetAll()
        {
            List<Account> accounts = await _accountRepository.GetAll();

            if (accounts != null)
            {
                return accounts.Select(accounts => _mapper.Map<Account, StaticAccountResponse>(accounts)).ToList();
            }
            return null;
        }

        public async Task<DirectAccountResponse?> GetById(int accountId)
        {
            Account account = await _accountRepository.GetById(accountId);

            if (account != null)
            {
                return _mapper.Map<DirectAccountResponse>(account);
            }
            return null;
        }

        public async Task<DirectAccountResponse> Update(int accountId, AccountRequest request)
        {
            Account account = await _accountRepository.Update(accountId, _mapper.Map<Account>(request));

            if (account != null)
            {
                return _mapper.Map<DirectAccountResponse>(account);
            }
            return null;
        }
    }
}

File: FoodHub-Web-Client/src/app/_services/account.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { DirectAccountResponse, StaticAccountResponse, AccountRequest } from '../_models/account';

@Injectable({
  providedIn: 'root'
})
export class AccountService {

  private url: string = environment.ApiUrl + "/Account";

  constructor(private http: HttpClient) { }

  public getAll(): Observable<StaticAccountResponse[]> {
    return this.http.get<StaticAccountResponse[]>(this.url);
  }

  public getById(accountId: number): Observable<DirectAccountResponse> {
    return this.http.get<DirectAccountResponse>(`${this.url}/${accountId}`);
  }

  public update(accountId: number, request: AccountRequest): Observable<DirectAccountResponse> {
    console.log(this.url, accountId, request);
    return this.http.put<DirectAccountResponse>(`${this.url}/${accountId}`, request);
  }

  public delete(accountId: number): Observable<DirectAccountResponse> {
    return this.http.delete<DirectAccountResponse>(`${this.url}/${accountId}`);
  }
}
